### PR TITLE
runelite-client: add chat recolor plugin

### PR DESCRIPTION
**Pull request by Adam-**

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java [hunk 1]
```diff
@@ -84,19 +90,354 @@ public void onResizableChanged(ResizeableChanged event)
...
+				ChatMessageType.FILTERED);
+		}
+		if (chatColorConfig.transparentFilteredHighlight() != null)
+		{
+			cacheColor(new ChatColor(ChatColorType.HIGHLIGHT, chatColorConfig.transparentFilteredHighlight(), true),
+				ChatMessageType.FILTERED);
+		}
+	}
+
 	public ChatMessageManager cacheColor(final ChatColor chatColor, final ChatMessageType... types)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java]: I would probably make this private now, so plugins will not be able to mess with it.

File: runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java [hunk 2]
```diff
@@ -84,19 +90,354 @@ public void onResizableChanged(ResizeableChanged event)
...
-			chatColors.add(chatColor);
+			// color is excluded from equals/hashCode on ChatColor
+			colorCache.remove(chatMessageType, chatColor);
+			colorCache.put(chatMessageType, chatColor);
 		}
 
 		return this;
 	}
 
+	public void clearColors()
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java]: This too

File: runelite-api/src/main/java/net/runelite/api/ChatMessageType.java [hunk 3]
```diff
@@ -124,10 +131,7 @@
 
 	private final int type;
 
-	ChatMessageType(int type)
-	{
-		this.type = type;
-	}
+	private static final Map<Integer, ChatMessageType> chatMessageTypes = new HashMap<>();
```
Comment (deathbeam) [inline runelite-api/src/main/java/net/runelite/api/ChatMessageType.java]: This is never filled, and also CHAT_MESSAGE_TYPES

File: runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java [hunk 4]
```diff
@@ -84,14 +90,392 @@ public void onResizableChanged(ResizeableChanged event)
...
+	}
+
+	private static Color getDefaultColor(ChatMessageType type, boolean transparent)
+	{
+		if (!transparent)
+		{
+			switch (type)
+			{
+				case PUBLIC:
+					return Color.decode("#0000FF");
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/chat/ChatMessageManager.java]: Maybe switching this other way around would reduce the code a bit, eg 1 switch and then return transparent ? : and default: return null

**Comment by Antedekebab** [comment]
Thanks aloot guys for the hard work you made me able to enjoy runescape again!!! <3

**Pull request merged**